lib/pdk/module/convert.rb

```diff
@@ -108,19 +108,22 @@ def template_url
 
       def update_metadata(metadata_path, template_metadata)
         if File.file?(metadata_path)
-          if File.readable?(metadata_path)
-            begin
-              metadata = PDK::Module::Metadata.from_file(metadata_path)
-              new_values = PDK::Module::Metadata::DEFAULTS.reject { |key, _| metadata.data.key?(key) }
-              metadata.update!(new_values)
-            rescue ArgumentError
-              metadata = PDK::Generate::Module.prepare_metadata(options) unless options[:noop] # rubocop:disable Metrics/BlockNesting
-            end
-          else
+          unless File.readable?(metadata_path)
             raise PDK::CLI::ExitWithError, _('Unable to update module metadata; %{path} exists but it is not readable.') % {
               path: metadata_path,
             }
           end
+
+          begin
+            metadata = PDK::Module::Metadata.from_file(metadata_path)
+            new_values = PDK::Module::Metadata::DEFAULTS.select do |key, _|
+              !metadata.data.key?(key) || metadata.data[key].nil? ||
```

Comment:
Hmm. This PR makes me think about how we handle `operatingsystem_support`. Currently.. if the key doesn't exist, we put the defaults in, which may be inaccurate. But if the key does exist and is empty, we leave it alone. With this PR, in both of the cases, we will update the metadata with the default operating system support, which might be undesirable.

This is out of scope of this PR, but IMO, I think we need to remove operating system support from the defaults, and only check the key (and prompt user to select OS) if `--full-interview` is specified or if they are doing a `pdk build` and we detect the key is missing.

Reply:
Yeah, I'm 👍 to replacing values that are `null` but I think replacing empty arrays with the defaults could be more problematic.

Reply:
Fair enough, but we'll have to add an exception for `requirements` as PDK will fail to validate as it expects to find a puppet version requirement.

Reply:
Should we just have a pre-condition somewhere that bails out if `requirements` is an empty array?

Reply:
I don't think so. If the requirements array exists and is empty, it is a pretty harmless change to add a puppet requirement and saves the user from having to make a manual change (either removing the array which would then be re-added with the default value anyway during convert, or manually creating the puppet requirement).

Reply:
Yeah, that sounds reasonable!